crates/wasi/src/preview2/host/udp.rs

```diff
@@ -34,8 +34,10 @@ impl<T: WasiView> udp::HostUdpSocket for T {
         local_address: IpSocketAddress,
     ) -> SocketResult<()> {
         let table = self.table_mut();
+        // Set the pool on the socket so later functions have access to it through the socket handle
+        table.get_mut(&this)?.pool = table.get(&network)?.pool.clone();
```

Comment:
Could this be done after the state validation below? That'd help prevent odd situations such as calling this multiple times just to switch the network while having the method return an error.

Reply:
@rylev  Don't know if this was your intention, but this creates a full copy of the entire pool for each socket.